sdk/core/core-amqp/src/util/checkNetworkConnection.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { resolve, CONNREFUSED, TIMEOUT } from "dns";
+import { retry as logRetry } from "../log";
+
+/**
+ * Checks whether a network connection is detected.
+ * @ignore
+ * @internal
+ */
+export function checkNetworkConnection(host: string): Promise<boolean> {
+  return new Promise((res) => {
+    logRetry("Calling dns.resolve to determine network connection status.");
+    resolve(host, function(err: any): void {
+      if (err) {
+        logRetry(
+          "Error thrown from dns.resolve in network connection check: '%s', %O",
+          err.code || err.name,
+          err
+        );
+        // List of possible DNS error codes: https://nodejs.org/dist/latest-v12.x/docs/api/dns.html#dns_error_codes
+        // Only when dns.resolve returns an error we expect to see when the network is down, resolve as 'false'.
+        if (err.code === CONNREFUSED || err.code === TIMEOUT) {
```

Comment:
Arent these suppose to be ECONNREFUSED and ETIMEOUT?

Reply:
These are now using the constants from the `dns` module. They are the same values as what we hard-coded as strings before.